### PR TITLE
added middleware for sending and handling notifications

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

File: ui/page.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const { merge } = require('lodash');
 const path = require('path');
 const findRoot = require('find-root');
 const { Router } = require('express');
+const notifications = require('../lib/middleware/notifications');
```
Comment (lennym) [inline ui/page.js]: This is on the page router and not global because it needs the content to have been set, right?

Comment (joefitter) [reply ui/page.js]: maybe, at least that was the intention, but it actually might not matter

Comment (joefitter) [reply ui/page.js]: as content is set in the global router, but it is extended here. But I guess at the point of calling the req.notification function the content will have been merged

Comment (lennym) [reply ui/page.js]: Yeah, that's fine. You _could_ work around it with some weird `onRequestEnd` stuff, but this is fine.

**Pull request merged**